File: app/src/main/java/de/fklappan/app/webnotes/ui/detail/DetailView.java
package de.fklappan.app.webnotes.ui.detail;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.leinardi.android.speeddial.SpeedDialView;

import de.fklappan.app.webnotes.R;
import de.fklappan.app.webnotes.common.mvx.BaseObservableMvxView;

public class DetailView extends BaseObservableMvxView<DetailContract.ViewListener> implements DetailContract.View {

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // member variables

    private TextView textViewTitle;
    private TextView textViewContent;
    private SpeedDialView fabView;

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // construction

    public DetailView(LayoutInflater inflater, ViewGroup viewGroup) {
        rootView = inflater.inflate(R.layout.detailview, viewGroup);
        initView();
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // member methods

    private void initView() {
        textViewTitle = rootView.findViewById(R.id.textViewTitle);
        textViewContent = rootView.findViewById(R.id.textViewContent);
        fabView = rootView.findViewById(R.id.fab);
        fabView.setOnChangeListener(new SpeedDialView.OnChangeListener() {
            @Override
            public boolean onMainActionSelected() {
                for (DetailContract.ViewListener listener : listeners) {
                    listener.onEditClicked();
                }
                return false;
            }

            @Override
            public void onToggleChanged(boolean isOpen) {

            }
        });
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // base class overrides

    @Override
    public View getRootView() {
        return rootView;
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // MvxView impl

    @Override
    public Bundle getViewState() {
        return null;
    }

    @Override
    public void cleanup() {

    }

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // DetailContract.View impl

    @Override
    public void setTitle(String title) {
        textViewTitle.setText(title);
    }

    @Override
    public void setContent(String content) {
        textViewContent.setText(content);
    }
}

File: app/src/main/java/de/fklappan/app/webnotes/ui/detail/DetailView.kt
package de.fklappan.app.webnotes.ui.detail

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import com.leinardi.android.speeddial.SpeedDialView

import de.fklappan.app.webnotes.R
import de.fklappan.app.webnotes.common.mvx.BaseObservableMvxView

class DetailView(inflater: LayoutInflater, viewGroup: ViewGroup?) : BaseObservableMvxView<DetailContract.ViewListener>(), DetailContract.View {

    private lateinit var textViewTitle: TextView
    private lateinit var textViewContent: TextView
    private lateinit var fabView: SpeedDialView

    init {
        rootView = inflater.inflate(R.layout.detailview, viewGroup)
        initView()
    }

    private fun initView() {
        textViewTitle = rootView.findViewById(R.id.textViewTitle)
        textViewContent = rootView.findViewById(R.id.textViewContent)
        fabView = rootView.findViewById(R.id.fab)
        fabView.setOnChangeListener(object : SpeedDialView.OnChangeListener {

            override fun onMainActionSelected(): Boolean {
                for(listener in listeners) {
                    listener.onEditClicked()
                }
                return false
            }

            override fun onToggleChanged(isOpen: Boolean) {

            }

        })
    }

    override fun getViewState(): Bundle? {
        return null
    }

    override fun cleanup() {
    }

    override fun setTitle(title: String) {
        textViewTitle.text = title
    }

    override fun setContent(content: String) {
        textViewContent.text = content
    }

    override fun startEditMode() {
    }

}
